Test feature store response

JoeDKoal94 opened the pull request:
Tests Store response feature. Do Pull requests for test-feature-survey-submission first.

cbosch1 committed:
Set up structure for Survey tasks

mcawful committed:
Get survey form by id tests

cbosch1 committed:
Updated getSurvey method name to proper name

Originally was called getQuestion

marcjacobroy committed:
implemented FormResponseServiceImpl and passed all tests

marcjacobroy committed:
implement SurveyResponseController

leonardoz15 committed:
renamed surveyResponseController and tests

decided to use formResponseController instead so it is consistant across the application

leonardoz15 committed:
refactored formResponse service

removed form response repo, implemented a new method in the messaging service to send a singular form response through the queue, need to update tests to reflect the change

leonardoz15 committed:
removed repo mocks from formResponseService tests

leonardoz15 committed:
added simple test for rabbitMQImpl

leonardoz15 committed:
completed testing for form response controller and service

jurczakn commented on SyncService/src/main/java/com/revature/controllers/FormResponseController.java:
```diff
@@ -0,0 +1,34 @@
+	 * @return ResponseEntity<Boolean>
+	 */
+	@PostMapping("/survey/response/{token}")
+	public ResponseEntity<Boolean> createFormResponse(@PathVariable("token")String token, @RequestBody FormResponseDto formResponseDto ) {
```
Handle with Status codes not a boolean

jurczakn commented on SyncService/src/main/java/com/revature/controllers/SurveyController.java:
```diff
@@ -0,0 +1,65 @@
+package com.revature.controllers;
```
Not Necessary

jurczakn commented on SyncService/src/main/java/com/revature/dto/FormResponseDto.java:
```diff
@@ -0,0 +1,163 @@
+	/**
+	 * variable of type {@link List}{@link String} that represents the questions of the response. 
+	 */
+	private List<String> questions;
```
consider a better way to store questions and answers

jurczakn commented on SyncService/src/main/java/com/revature/models/FormResponse.java:
```diff
@@ -67,7 +67,7 @@ public void setFormId(int formId) {
 	/**
 	 * Gets the time stamp for the form.
-	 * @return form time stamp
+	 * @return time stamp time stamp
```
maybe fix

jurczakn commented on SyncService/src/test/java/com/revature/controllers/FormResponseControllerTest.java:
```diff
@@ -0,0 +1,122 @@
+	 * will be returned by the controller along with a BadRequest HttpStatus code.
+	 */
+	@Test
+	void testCreateSurveyResponse_withErrors() {
```
There could be more reasons why this would fail. Break into multiple tests

1. failed to auth JWT
2. if survey has already been submitted
3. null values
4. missing answers

jurczakn commented on SyncService/src/test/java/com/revature/service/FormResponseServiceTest.java:
```diff
@@ -0,0 +1,152 @@
+
+		FormResponse returned = service.createFormResponse(formResponse, token);
+
+		verify(authService).verifyJWT(token);
```
verify the response is not saved

jurczakn commented on SyncService/src/test/java/com/revature/service/FormResponseServiceTest.java:
```diff
@@ -0,0 +1,152 @@
+		verify(authService).verifyJWT(token);
+		verify(authService).getClaim();
+		verify(associateSurveySessionService).readAssociateSurveySession((int) claims.get("surveySubId"));
+
```
check not saved

jurczakn commented on SyncService/src/test/java/com/revature/service/FormResponseServiceTest.java:
```diff
@@ -0,0 +1,152 @@
+		verify(authService).getClaim();
+		verify(associateSurveySessionService).readAssociateSurveySession((int) claims.get("surveySubId"));
+		verify(associateSurveySessionService).updateAssociateSurveySession(updatedAssociateSurveySession);
+
```
check not saved

jurczakn commented on SyncService/src/test/java/com/revature/service/FormResponseServiceTest.java:
```diff
@@ -0,0 +1,152 @@
+package com.revature.service;
```
test the answers match questions on the survey

jurczakn reviewed (changes requested):
Remove .DS_Store files

leonardoz15 committed:
Merge pull request #19 from Batch-898-NickJurczak/test-feature-verify-token

Test feature verify token

leonardoz15 committed:
implemented most tests for FormResponseController

leonardoz15 committed:
completed testing for form response controller/services

leonardoz15 committed:
updated formResponseDto and controller

leonardoz15 committed:
Merge pull request #22 from Batch-898-NickJurczak/test-feature-get-survey

Test feature get survey

leonardoz15 committed:
refactored FormResponseDto, fixed controller and service tests

Added another test to check for invalid questions and answers along with a custom exception

leonardoz15 committed:
deleted sync-errors

leonardoz15 committed:
Merge branch 'feature-store-response' into test-feature-store-response

leonardoz15 committed:
removed implementation

leonardoz15 committed:
updated tests and formRespones models

leonardoz15 committed:
Delete .DS_Store

leonardoz15 committed:
fixed merge

leonardoz15 committed:
fixed merge conflicts

leonardoz15 committed:
Merge branch 'dev' into test-feature-store-response
